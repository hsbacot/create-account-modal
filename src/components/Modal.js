import React from "react";
import PropTypes from "prop-types";

const Modal = ({ children }) => (
  <div className="modal-wrapper">
    <div className="modal-content">
      {children}
    </div>
  </div>
);

Modal.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element])
};

export default Modal;
