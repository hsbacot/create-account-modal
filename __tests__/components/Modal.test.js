import React from "react";
import { shallow } from "enzyme";
import Modal from "../../src/components/Modal";

describe("<Modal />", () => {
  it("should render", () => {
    const component = shallow(<Modal />);
    expect(component).toMatchSnapshot();
  });

  it("should render children", () => {
    const component = shallow(
      <Modal>
        <h1>children</h1>
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });
});
