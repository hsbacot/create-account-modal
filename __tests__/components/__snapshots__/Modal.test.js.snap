exports[`<Modal /> should render 1`] = `
<div
  className="modal-wrapper">
  <div
    className="modal-content" />
</div>
`;

exports[`<Modal /> should render children 1`] = `
<div
  className="modal-wrapper">
  <div
    className="modal-content">
    <h1>
      children
    </h1>
  </div>
</div>
`;
